### PR TITLE
Add in a lookuptable thumbnail function

### PR DESCRIPTION
**Pull request by alexgleith**
Add in a way to call the existing function for a singleband lookuptable from the dataset assembler class.

### CODE REVIEW

**Review by omad — DISMISSED** [review]
Looks like an improvement to me, pushing more code in here from `datacube-alchemist`.

My only concern, and not a blocker, is that this doesn't go far enough. It's still too method call driven which means extra logic in the client. Maybe we should add parameters to `DatasetAssembler.write_thumbnail()` and it gets to decide on the method to use to generate the thumbnail...

**Comment by alexgleith** [comment]
What do you think, @jeremyh? Want me to weave the two functions into a single meta function with two code paths?

**Review by omad — DISMISSED** [review]

File: eodatasets3/assemble.py [hunk 1]
```diff
@@ -846,12 +857,52 @@ def write_thumbnail(
...
-        accessory_name = "thumbnail"
-        if kind:
-            accessory_name += f":{kind}"
-        self.add_accessory_file(accessory_name, thumb_path)
+        self._document_thumbnail(thumb_path, kind)
+
+    def write_thumbnail_singleband(
+        self,
+        measurement: str,
+        thumb_path,
```
Comment (jeremyh) [inline eodatasets3/assemble.py]: This argument may have been for debugging? (it's never used, being overridden in the first line. And the paired `create_thumbnail` function doesn't have it either)

File: eodatasets3/assemble.py [hunk 2]
```diff
@@ -846,12 +857,52 @@ def write_thumbnail(
...
+        Use a bit (int) as the value to scale from black to white to
+        i.e., 0 will be BLACK and bit will be WHITE, with a linear scale between.
+        OR
+        Provide a lookuptable (dict) of int (key) [R, G, B] (value) fields
+        to make the image with.
+        """
+
+        thumb_path = self.names.thumbnail_name(self._work_path, kind=kind)
+
+        _, image_path = self.measurements.get(measurement, None)
```
Comment (jeremyh) [inline eodatasets3/assemble.py]: The unpacking will always fail when measurement is unknown, so we'll never reach the below validation.

```suggestion
        _, image_path = self.measurements.get(measurement, (None, None))
```

**Review by jeremyh — COMMENTED** [review]
Looks good. It's different enough of a situation that I don't mind a second function for convenience. Though I like where Damien's coming from.

If combining them means making all the current red/green/blue args optional on the existing function I'd probably prefer not to combine, as their arguments are too different and we'll have a confusing set of completely-mutually-exclusive optional/required parameter combinations on the one function.

--- 

Perhaps also add a line to one of our minimal tests to run the function, eg:

```
p.write_thumbnail_singleband('blue', None, bit=1)
```
to https://github.com/GeoscienceAustralia/eo-datasets/blob/224f97410995e083982fa20ce75fe8e069f8fdbf/tests/integration/test_assemble.py#L200

(I also ran it at first without any `bit` argument and got a cryptic error message -- we're never checking the `bit is None and lookup_table is None` case in our old method parameter checking)

**Comment by jeremyh** [comment]
Bike shedding... should it be `write_thumbnail_singleband` or `write_thumbnail_single_band` to have consistent naming style? (we have, for example `self.write_measurements_odc_xarray()` elsewhere)

**Comment by alexgleith** [comment]
> Bike shedding... should it be `write_thumbnail_singleband` or `write_thumbnail_single_band` to have consistent naming style? (we have, for example `self.write_measurements_odc_xarray()` elsewhere)

I think singleband is a single word... keeping it as it is ;-)

**Comment by omad** [comment]
Install `pre-commit` Alex!

**Review by omad — APPROVED** [review]
👯

**Review by jeremyh — APPROVED** [review]
👍

**Pull request merged**